### PR TITLE
Define ``generators`` on ``ComprehensionScope``

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] Write a good description on what the PR does.

## Description

Ref https://github.com/PyCQA/astroid/pull/1475.

This is necessary for `ComprehensionScope` to be able to access `generators` without `mypy` complaining. It doesn't really change behaviour either way and actually fixes an unintentional bug where `ListComp.generators == None` even though `postinit` is called.
_I see no reason why that behaviour shouldn't be the same as for all other `generators` attributes._

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Related Issue

### CODE REVIEW

File: astroid/nodes/scoped_nodes/scoped_nodes.py [hunk 1]
```diff
@@ -724,14 +722,13 @@ def __init__(
             parent=parent,
         )
 
-    def postinit(self, elt=None, generators=None):
+    def postinit(self, elt=None, generators: Optional["nodes.Comprehension"] = None):
         """Do some setup after initialisation.
 
         :param elt: The element that forms the output of the expression.
         :type elt: NodeNG or None
```
Comment (Pierre-Sassoulas) [inline astroid/nodes/scoped_nodes/scoped_nodes.py:729]:
```suggestion
    def postinit(self, elt: "nodes.NodeNG" =None, generators: Optional["nodes.Comprehension"] = None):
        """Do some setup after initialisation.

        :param elt: The element that forms the output of the expression.
```
Wondering also if it should be ``Optional[List["nodes.Comprehension"]]``

File: astroid/nodes/scoped_nodes/scoped_nodes.py [hunk 2]
```diff
@@ -822,7 +814,9 @@ def __init__(
             parent=parent,
         )
 
-    def postinit(self, key=None, value=None, generators=None):
+    def postinit(
+        self, key=None, value=None, generators: Optional["nodes.Comprehension"] = None
```
Comment (Pierre-Sassoulas) [inline astroid/nodes/scoped_nodes/scoped_nodes.py:818]:
```suggestion
        self, key=None, value=None, generators: Optional[List["nodes.Comprehension"]] = None
```
?

**Review by Pierre-Sassoulas — COMMENTED** [review]
I did not comment on all the Optional[nodes.Comprehension], I suppose I'm missing something.

File: astroid/nodes/scoped_nodes/scoped_nodes.py (hunk 1 above)
Comment (DanielNoord) [reply astroid/nodes/scoped_nodes/scoped_nodes.py]: Well, that was quite stupid... Fixed!

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**